src/box/iproto.cc

```diff
@@ -2704,7 +2746,88 @@ iproto_req_handler_is_set(struct iproto_req_handlers *handlers)
 	if (handlers == NULL)
 		return false;
 
-	return handlers->c.cb != NULL;
+	return handlers->event_by_id != NULL ||
+	       handlers->event_by_name != NULL ||
+	       handlers->c.cb != NULL;
+}
+
+/**
+ * Returns `enum iproto_type' if `name' is a valid IPROTO type name or equals
+ * "unknown". Otherwise, iproto_type_MAX is returned. The name is expected to
+ * be in lowercase.
+ */
+static enum iproto_type
+get_iproto_type_by_name(const char *name)
+{
+	enum iproto_type type = iproto_type_MAX;
+	for (uint32_t i = 0; i < iproto_type_MAX; i++) {
+		const char *type_name = iproto_type_name(i);
+		if (type_name == NULL)
+			continue;
+		char *type_name_lower = strtolowerdup(type_name);
+		if (strcmp(type_name_lower, name) == 0) {
+			type = (enum iproto_type)i;
+			free(type_name_lower);
+			break;
+		}
+		free(type_name_lower);
```

Comment:
> We will now have 2 places where we do this lowercase conversion (see also src/box/lua/iproto.c).

In `src/box/lua/iproto.c` the `iproto_key_strs` is lowered, not `iproto_type_strs`.

> How about creating an array of lowercase string constants one time during iproto initialization?

Done.

> That way you also use strindex from src/lib/core/utils.c here.

`strindex()` ignores the case of the characters, but I need case-sensitive comparison here, because from the trigger-registry point of view the events "box.iproto.override.ping" and "box.iproto.override.PING" are totally unique. For this reason only lowercase names are allowed in `get_iproto_type_by_name()` to avoid any further inconsistency.